
# Integration 
- [ ] [Computational principles and challenges in single-cell data integration](https://www.nature.com/articles/s41587-021-00895-7)
- [ ] [A benchmark of batch-effect correction methods for single-cell RNA sequencing data
](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1850-9)
- [ ] [Benchmarking atlas-level data integration in single-cell genomics](https://www.nature.com/articles/s41592-021-01336-8)  

# Spatial/scRNASeq
- [ ] [Reconstruction of the cell pseudo-space from single-cell RNA sequencing data with scSpace](https://www.nature.com/articles/s41467-023-38121-4)
- [ ] [Optimizing Xenium In Situ data utility by quality assessment and best practice analysis workflows](https://www.biorxiv.org/content/10.1101/2023.02.13.528102v1.full.pdf)
- [ ] https://www.biorxiv.org/content/10.1101/2022.10.06.510405v1

- [ ] [bioimage model zoo](https://bioimage.io/#/)
- [ ] [BioImage Model Zoo: A Community-Driven Resource for Accessible Deep Learning in BioImage Analysis](https://www.biorxiv.org/content/10.1101/2022.06.07.495102v1)

- [ ] https://ai4life.eurobioimaging.eu/
- [ ] [Whole-body integration of gene expression and single-cell morphology](https://www.sciencedirect.com/science/article/pii/S009286742100876X)
- [ ] [confronting false discoveries in single-cell differential expression](https://www.nature.com/articles/s41467-021-25960-2)
- [ ] [Seurat](https://satijalab.org/seurat/)  
- [ ] [SingleCellExperiment](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html)  
- [ ] [Monocole](https://cole-trapnell-lab.github.io/monocle-release/)
- [ ] [Scanpy](https://scanpy.readthedocs.io/en/stable/)  
 
